inspirehep/modules/hal/tei.py

```diff
@@ -0,0 +1,223 @@
+
+    for record in records:
+        structures.append(
+            _structure_data(record)
+        )
+    return dedupe_list(structures)
+
+
+def _parse_pub_info(record):
+    try:
+        pub_info = _lookup(record, "publication_info[0]")
+        if 'journal_title' in pub_info:
+            return _journal_data(pub_info)
+        elif 'conference_record' in pub_info:
+            return _conference_data(pub_info['conference_record'])
+        else:
+            return None
+    except KeyError:
+        return None
+
+
+def _parse_dois(record):
+    return get_value(record, "dois[0].value", "")
+
+
+def _parse_authors(record):
+    authors = get_value(record, "authors", [])
+    for author in authors:
+        try:
+            parsed = HumanName(author['full_name'])
+            author['parsed_name'] = parsed
+        except KeyError:
+            continue
+    return authors
+
+
+def _conference_data(conf):
+    ref = replace_refs(conf, 'db')
+
+    # FIXME: Add conference city, country, and country code fields
+    if ref:
+        return {'type': "conference",
+                'name': get_value(ref, "titles[0].title", ""),
+                'acronym': get_value(ref, "acronym[0]", ""),
+                'opening_date': get_value(ref, "opening_date", ""),
+                'closing_date': get_value(ref, "closing_date", "")}
+    else:
+        return {'type': "conference",
+                'name': "",
+                'acronym': "",
+                'opening_date': "",
+                'closing_date': ""}
+
+
+def _journal_data(pub_info):
+    if 'page_artid' in pub_info:
+        pp = pub_info['page_artid']
+    elif 'page_start' in pub_info and 'page_end' in pub_info:
+        pp = pub_info['page_start'] + "-" + pub_info['page_end']
+    elif 'page_start' in pub_info or 'page_end' in pub_info:
+        pp = pub_info['page_start'] or pub_info['page_end']
+    else:
+        pp = ""
+
+    return {
+        'type': "journal",
+        'name': pub_info.get('journal_title', ""),
+        'year': pub_info.get('year', ""),
+        'volume': pub_info.get('journal_volume', ""),
+        'issue': pub_info.get('journal_issue', ""),
+        'pp': pp,
+    }
+
+
+def _structure_data(struct):
+    return {
+        'type': get_value(struct, "collections[1].primary", "").lower(),
+        # ^^ FIXME: This may not be one of the HAL accepted values:
+        # institution, department, laboratory or researchteam
+        'name': get_value(struct, "institution[0]", ""),
+        'address': get_value(struct, "address[0].original_address", []),
+        'country': get_value(struct, "address[0].country_code", ""),
+        'recid': get_recid_from_ref(struct['self']),
+    }
+
+
+def _lookup(record, value):
+    """Searches a key in a record.
+
+    Uses `get_value` to lookup a key in a JSON record, and
+    raises a `KeyError` it it wasn't found.
+    """
+    result = get_value(record, value)
+    if not result:
+        raise KeyError
+    return result
```

Comment:
Yeah, it's necessary to allow me to pass the return value to other functions that don't handle None or empty lists.